tests/suites/test_suite_psa_crypto.function

```diff
@@ -9685,6 +9685,9 @@ void generate_key_rsa(int bits_arg,
     }
 
     /* Test the key information */
+#if (defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_RSA_KEY_PAIR_IMPORT) && \
+    defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_RSA_KEY_PAIR_EXPORT)) || \
+    defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_RSA_PUBLIC_KEY)
```

Comment:
Question: it's not obvious to me why you picked those exact guards, can you explain?

Reply:
These macros come from:
https://github.com/Mbed-TLS/mbedtls/blob/f3ccfddb45dc964124145789e433f22d382ca5a6/library/psa_crypto.c#L1375-L1405

If these conditions are not fitted, `psa_get_key_attributes` can't get RSA key attributes but returns `PSA_SUCCESS`.

Reply:
Thanks for the explanation! I'll note the comment there is outdated (references an issue that was closed) and I'm tempted to slightly expand the scope of that PR to include updating that comment, but I'm not sure what the proper replacement would be yet.

@gilles-peskine-arm It looks like the prediction in [the last comment in the issue listed above](https://github.com/ARMmbed/mbed-crypto/issues/216#issuecomment-1561795353) was not realised, as we seem to now find ourselves in a situation where we have an (experimental, in `crypto_extra.h`) API for custom domains parameters, with `psa_set/get_key_domain_parameters()` allowing to set/get custom public exponents for RSA keys, but no corresponding support in the driver API that I could find.

More precisely, I'm not sure how `psa_get_key_attributes()` is supposed to populate the attributes structure with the public exponent in case RSA keys are supported only by a driver? I think it's OK if we have a gap for now (which could be closed either by expanding driver support or by removing the experimental `domain_parameters` extension), but then we need to make sure we have an open issue tracking it.

Also, if we accept this gap for now, I'm a bit uncomfortable with the fact that `psa_get_key_attributes()` and then `psa_get_key_domain_parameters()` will both return success even when the actual value of `e` can't be accessed. I'd say this would deserve at least a warning in the documentation of `psa_get_key_domain_parameters()` that with RSA keys handled by a driver, the exponent will always be reported as the default one regardless of its actual value. Or should we make `psa_get_key_domain_parameters()` return an error in this case?

Reply:
We're in the uncomfortable position where domain parameters are deprecated, so we haven't included them in newer parts of the library, such as the driver interface. But we haven't removed them partly because removing takes time and partly because there is one thing that we care about that requires them, namely generating an RSA key with a non-default exponent.

My preferred resolution would be to:

1. Finish [designing `psa_generate_key_ext`](https://github.com/Mbed-TLS/mbedtls/issues/6494) and implement it and its driver counterpart.
2. In a subsequent minor release, remove domain parameters.
3. Now you don't have this problem anymore.

But obviously that would take time that we don't have.

In the meantime, I think ideally `psa_get_key_domain_parameters` should return an error if it can't obtain the public exponent. `psa_get_key_attributes` must always succeed (except if access to the key object fails in some way, e.g. an invalid key identifier or an operational error).